core/proofs/src/circuit_transfer.rs

```diff
@@ -342,62 +391,67 @@ fn u32_into_boolean_vec_le<E, CS>(
                 )?))
             })
             .collect::<Result<Vec<_>, SynthesisError>>()?;
-    
+
     Ok(bits)
 }
 
 #[cfg(test)]
 mod tests {
     use super::*;
     use pairing::{bls12_381::{Bls12, Fr}, Field};
-    use rand::{SeedableRng, Rng, XorShiftRng, Rand};    
+    use rand::{SeedableRng, Rng, XorShiftRng, Rand};
     use crate::circuit_test::TestConstraintSystem;
-    use scrypto::jubjub::{JubjubBls12, fs, edwards, JubjubParams};    
-    use crate::primitives::EncryptionKey;       
-    
+    use scrypto::jubjub::{JubjubBls12, fs, edwards, JubjubParams};
+    use crate::primitives::EncryptionKey;
+
     #[test]
-    fn test_circuit_transfer() {        
+    fn test_circuit_transfer() {
         let params = &JubjubBls12::new();
-        let rng = &mut XorShiftRng::from_seed([0x3dbe6258, 0x8d313d76, 0x3237db17, 0xe5bc0654]);        
+        let rng = &mut XorShiftRng::from_seed([0x3dbe6258, 0x8d313d76, 0x3237db17, 0xe5bc0654]);
 
         let sk_fs_s: fs::Fs = rng.gen();
         let sk_fs_r: fs::Fs = rng.gen();
 
         let proof_generation_key_s = ProofGenerationKey::<Bls12>::from_origin_key(&sk_fs_s, params);
         let proof_generation_key_r = ProofGenerationKey::<Bls12>::from_origin_key(&sk_fs_r, params);
-        
-        let decryption_key_s: fs::Fs = proof_generation_key_s.bdk();                                             
-        let decryption_key_r: fs::Fs = proof_generation_key_r.bdk();                                                             
+
+        let decryption_key_s: fs::Fs = proof_generation_key_s.bdk();
+        let decryption_key_r: fs::Fs = proof_generation_key_r.bdk();
 
         let address_recipient = EncryptionKey::from_origin_key(&sk_fs_r, params);
-        let address_sender_xy = proof_generation_key_s.into_encryption_key(params).0.into_xy(); 
-        let address_recipient_xy = address_recipient.0.into_xy();                       
-        
+        let address_sender_xy = proof_generation_key_s.into_encryption_key(params).0.into_xy();
+        let address_recipient_xy = address_recipient.0.into_xy();
+
         let alpha: fs::Fs = rng.gen();
 
         let value = 10 as u32;
         let remaining_balance = 17 as u32;
```

Comment:
You need to fix here:
`let remaining_balance = 16 `

Reply:
I overlooked that!

Reply:
OMG my code pass the test!